### ISSUE TITLE
[MNG-7676] On Maven 3.9.0+ release, sha512 hashes have "null" classifier

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#8776]
**[Tamas Cservenak](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=cstamas)** opened **[MNG-7676](https://issues.apache.org/jira/browse/MNG-7676?redirect=false)** and commented

We noticed this (Guillaume for 4.0.0-alpha releases) and for 3.9.0 release as well, the generated SHA512 hash had "null" classifier.

```
apache-maven-3.9.0-src.tar.gz    Tue Jan 31 10:00:59 UTC 2023    2781108    
apache-maven-3.9.0-src.tar.gz.asc    Tue Jan 31 10:01:01 UTC 2023    833    
apache-maven-3.9.0-src.tar.gz.md5    Tue Jan 31 10:01:00 UTC 2023    32    
apache-maven-3.9.0-src.tar.gz.sha1    Tue Jan 31 10:01:00 UTC 2023    40    
apache-maven-3.9.0-src.zip    Tue Jan 31 10:00:59 UTC 2023    4649980    
apache-maven-3.9.0-src.zip.asc    Tue Jan 31 10:01:00 UTC 2023    833    
apache-maven-3.9.0-src.zip.md5    Tue Jan 31 10:00:59 UTC 2023    32    
apache-maven-3.9.0-src.zip.sha1    Tue Jan 31 10:00:59 UTC 2023    40    
apache-maven-3.9.0.null.sha512    Tue Jan 31 10:01:00 UTC 2023    128    
apache-maven-3.9.0.null.sha512.md5    Tue Jan 31 10:01:00 UTC 2023    32    
apache-maven-3.9.0.null.sha512.sha1    Tue Jan 31 10:01:00 UTC 2023    40    
apache-maven-3.9.0.pom    Tue Jan 31 10:00:53 UTC 2023    10140    
apache-maven-3.9.0.pom.asc    Tue Jan 31 10:01:00 UTC 2023    833    
apache-maven-3.9.0.pom.md5    Tue Jan 31 10:00:56 UTC 2023    32    
apache-maven-3.9.0.pom.sha1    Tue Jan 31 10:00:56 UTC 2023    40     
```

During release this was plugin output (really says it will do this above):

```
[INFO] --- checksum-maven-plugin:1.11:files (source-release-checksum) @ apache-maven ---
[INFO] apache-maven-3.9.0-bin.zip - SHA-512 : 564fe44bfa9c7ad3e2703cbbac59d43a11fa39e4e68875d3d1584d0a0b7b77a1352da246b875c4c15d11ceb6b4dd9a0ce7dd7a48695725dce594f34325c9c605
[INFO] apache-maven-3.9.0-bin.tar.gz - SHA-512 : 1ea149f4e48bc7b34d554aef86f948eca7df4e7874e30caf449f3708e4f8487c71a5e5c072a05f17c60406176ebeeaf56b5f895090c7346f8238e2da06cf6ecd
[WARNING] artifact org.apache.maven:apache-maven:null.sha512:3.9.0 already attached, replace previous instance
[INFO] apache-maven-3.9.0-src.zip - SHA-512 : 68b1a3cde17a42814ea28499e0c3f5cb98c343cebdfb96b52ba5acc6c2c6bac4f99dcf93177840bbc4d2b32668f73315f678ba0fef211b422303a751690a139c
[WARNING] artifact org.apache.maven:apache-maven:null.sha512:3.9.0 already attached, replace previous instance
[INFO] apache-maven-3.9.0-src.tar.gz - SHA-512 : c9c89d49aa577223b1358f7391859535be031201eab502e0620522ceb66f2b42f6f8361764d6d627b66f6ae8c61cfe85cb51f7fb734aadd0427f10e43d5a1753
[WARNING] artifact org.apache.maven:apache-maven:null.sha512:3.9.0 already attached, replace previous instance
```

Seems we use wrong goal, files vs artifacts? At least ASF Parent POM shows a change for source releases, but Maven does not reuse it?

---

**Remote Links:**
- [GitHub Pull Request #977
](https://github.com/apache/maven/pull/977)

**Backported to:** [4.0.0-alpha-5](https://github.com/apache/maven/milestone/4?closed=1), [3.9.1](https://github.com/apache/maven/milestone/6?closed=1), [3.8.9](https://github.com/apache/maven/milestone/110?closed=1)

**Comment by jira-importer** [comment]
**[Tamas Cservenak](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=cstamas)** commented

3.9.x https://github.com/apache/maven/commit/d7e655348a3d9e32835184d68791420b7f22b845

master https://github.com/apache/maven/commit/489e4e6526ba0012113642252cc6c16e423720dc

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [7959f0df07fa94788b7d3d838033d44c754ea4e4](https://gitbox.apache.org/repos/asf?p=maven.git;a=commit;h=7959f0df07fa94788b7d3d838033d44c754ea4e4) for `maven-3.8.x` branch.
